framework/src/constraints/ElemElemConstraint.C

```diff
@@ -0,0 +1,142 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ElemElemConstraint.h"
+#include "FEProblem.h"
+#include "Assembly.h"
+
+// libMesh includes
+#include "libmesh/quadrature.h"
+
+template<>
+InputParameters validParams<ElemElemConstraint>()
+{
+  InputParameters params = validParams<Constraint>();
+  params.addParam<unsigned int>("interface_id", 0, "The id of the interface.");
```

Comment:
For the time being, it can't be anything other than 0. Down the road, this will allow us to define different behavior for different interfaces, but we currently don't have any way to associate a set of cut element pairs to an interface id.